Classes/Notifications/NotificationSectionController.swift

```diff
@@ -45,20 +45,20 @@ SwipeCollectionViewCellDelegate {
             let cell = collectionContext?.cellForItem(at: index, sectionController: self) as? NotificationCell
             else { fatalError("Missing object, cell missing, or incorrect type") }
 
-        if NotificationClient.readOnOpen() {
-            cell.isRead = true
-            client.markNotificationRead(id: object.id, isOpen: true)
-        }
-
         switch object.identifier {
         case .hash(let hash):
+            if NotificationClient.readOnOpen() {
+                cell.isRead = true
+                client.markNotificationRead(id: object.id, isOpen: true)
+            }
             viewController?.presentCommit(owner: object.owner, repo: object.repo, hash: hash)
         case .number(let number):
             let model = IssueDetailsModel(owner: object.owner, repo: object.repo, number: number)
             let controller = IssuesViewController(
                 client: client.githubClient,
                 model: model,
-                scrollToBottom: true
+                scrollToBottom: true,
+                issueDownloaded: markRead
```

Comment:
`markRead` itself doesn't update `cell.isRead` and also doesn't check if `NotificationClient.readOnOpen()` is true (it's a user-configurable option).

An interim block is needed to check these values before forwarding the request 😄

Reply:
Yes, checking on `NotificationClient.readOnOpen()` I have missed. And I guessed that combination of  `client.markNotificationRead(id: object.id, isOpen: false)` and `context.reload(self)` shoud update `cell.isRead`. Also I haven not access to cell in the closure.

Reply:
You should be able to just do `self?.cell.isRead = true` as per the original block - Obviously ensuring you don't hold onto `self` as to hold a retain cycle! Not sure that's a safe assumption given the code was added before, regardless you can't view the cell anyway so it may not be important as when you navigate back it will likely update everything from cache anyway. - Needs testing though.

Reply:
@Sherlouk I believe this will be right 😀  Btw, I can view the updating cell in case of iPad split view